packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -295,19 +295,17 @@ export class CdkToolkit {
     // |            |                |      | <------------------ |            | <------------------ |            | <-------------|
     // --------------                --------  'cdk deploy' done  --------------  'cdk deploy' done  --------------
     let latch: 'pre-ready' | 'open' | 'deploying' | 'queued' = 'pre-ready';
-    chokidar.watch(watchIncludes, {
-      ignored: watchExcludes,
-      cwd: rootDir,
-      // ignoreInitial: true,
-    }).on('ready', () => {
-      latch = 'open';
-      debug("'watch' received the 'ready' event. From now on, all file changes will trigger a deployment");
-    }).on('all', async (event, filePath) => {
+
+    const workflow = async (event: 'add' | 'addDir' | 'change' | 'unlink' | 'unlinkDir', filePath: string) => {
       if (latch === 'pre-ready') {
         print(`'watch' is observing ${event === 'addDir' ? 'directory' : 'the file'} '%s' for changes`, filePath);
       } else if (latch === 'open') {
         latch = 'deploying';
-        print("Detected change to '%s' (type: %s). Triggering 'cdk deploy'", filePath, event);
+        if (filePath === 'initial') {
```

Comment:
I don't love this as I'm hijacking `filePath` to pass in this `'initial'` flag, though I don't know what else to do since we want to bypass the print statement.

Reply:
I'm wondering. Can you try making `filePath` optional? (`filePath: string | undefined`) And differentiate the initial case that way, instead of the magic value `'initial'`.

Can you check whether that compiles?

Reply:
works, updated